crates/lumina_server/src/lobby/in_game.rs

```diff
@@ -45,69 +44,48 @@ fn start_game(
     }
 }
 
-/// This function updates the position of newly spawned spaceships
+/// Updates the position of newly spawned spaceships based on their assigned team type.
 fn init_spaceship_position(
     mut commands: Commands,
-    mut q_spaceships: Query<
-        (&mut Position, &mut Rotation, &PlayerId, Entity),
-        (
-            With<Spaceship>,
-            With<SourceEntity>,
-            Without<PositionInitialized>,
-        ),
-    >,
-    q_in_game_lobbies: Query<(), With<LobbyInGame>>,
+    mut spaceship_query: Query<(&mut Position, &PlayerId, &TeamType, Entity), With<Spaceship>>,
+    in_game_lobbies_query: Query<(), With<LobbyInGame>>,
     terrain_config: TerrainConfig,
-    lobby_infos: Res<LobbyInfos>,
-    mut team_toggle: ResMut<TeamToggle>, // Access the team toggle resource
+    lobby_info: Res<LobbyInfos>,
 ) {
+    // Ensure the terrain config is available
     let Some(terrain_config) = terrain_config.get() else {
         eprintln!("Terrain config is not available!");
```

Comment:
```suggestion
        error!("Terrain config is not available!");
```

`error!` or `warn!` is the preferred logging macros for bevy apps.